platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/exec/RunAsBuildOperationBuildActionExecutor.java

```diff
@@ -79,8 +76,8 @@ public BuildActionRunner.Result call(BuildOperationContext buildOperationContext
                 public BuildOperationDescriptor.Builder description() {
                     return BuildOperationDescriptor.displayName("Run build").details(new RunBuildBuildOperationType.Details() {
                         @Override
-                        public Map<Throwable, Collection<Problem>> getProblemsForThrowables() {
-                            return problemContainer.getProblemsForThrowables();
+                        public ProblemLookup getProblemLookup() {
+                            return problemContainer.getProblemLookup();
```

Comment:
🤔 That looks somewhat problematic: The problem lookup is only ready at the end, though we allow the lookup from the details object. Why is the problem lookup not a service?

Reply:
I'll probably need some help with that. FWIW problemlookup needs to be accessible ClientForwardingBuildOperationListener which is created in the context of `ToolingBuilderServices` which has no access to build tree scope services.